examples/post-sync/main.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"kpt.dev/configsync/pkg/api/configsync/v1beta1"
+)
+
+var (
+	runtimeScheme = runtime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(runtimeScheme))
+	utilruntime.Must(v1beta1.AddToScheme(runtimeScheme))
+}
+
+// run initializes and starts the controller manager with RootSync and RepoSync controllers.
+// It returns an error if any step fails.
+func run() error {
+	zapLog, err := zap.NewProduction()
```

Comment:
Why are we using zap and not klog?

The rest of our code uses klog i think. See the reconciler cmd.

Reply:
Zap generates logs in JSON format, which offers better parsing capabilities for downstream processing, in components like Cloud Functions. The initial implementation actually used klog, which presented challenges in processing and validation due to its less structured format. While it's true that Zap's JSON logging consumes more storage space compared to klog's string format, I recommend maintaining the JSON format for now. This approach provides greater flexibility and reliability for integration purposes. We can revisit this decision and potentially switch to a klog-compatible solution once the component is reintegrated into Config Sync.

Reply:
fair enough